### PR TITLE
Allow passing a function to match the origin

### PR DESCRIPTION
**Pull request by msvab**
Hello,

This is a great library! But we are running our API on two domains, which currently cannot be satisfied with the API, so this change introduces a function that can be used to match the origin.

### CODE REVIEW

**Comment by bradvogel** [comment]
@skeggse can this be merged?

**Comment by bradvogel** [comment]
Closing due to age.

**Review by skeggse — COMMENTED** [review]
oh hah i don't remember seeing this but i do like this. we should :shipit:

**Comment by bradvogel** [comment]
@msvab if this is still need, mind fixing the merge conflict?

File: index.js
Comment (skeggse) [inline index.js:52]:
```suggestion
    (origin) => origin === options.origin.toLowerCase();
```

**Comment by msvab** [comment]
> @msvab if this is still need, mind fixing the merge conflict?

I've updated the PR, let me know if any more changes are needed

File: index.js
Comment (skeggse) [inline index.js:57]:
```suggestion
    allowOrigin = (origin) => origin === options.origin.toLowerCase();
```

**Review by skeggse — APPROVED** [review]

**Comment by skeggse** [comment]
@bradvogel this lgtm; I don't have merge access anymore, so your call.

**Review by bradvogel — APPROVED** [review]

**Pull request merged**
